src/features/ExtensionCommands.ts

```diff
@@ -265,6 +270,16 @@ export class ExtensionCommandsFeature implements IFeature {
             this.languageClient.onRequest(
                 SetStatusBarMessageRequestType,
                 (messageDetails) => this.setStatusBarMessage(messageDetails));
+
+            this.languageClient.onNotification(
+                ClearTerminalNotificationType,
+                () => {
+                    // We check to see if they have TrueClear on. If not, no-op because the
+                    // overriden Clear-Host already calls [System.Console]::Clear()
```

Comment:
If `Clear` just triggered `\x1b[3J` it would work everywhere as that clears scrollback.

https://github.com/xtermjs/xterm.js/blob/70babeacb62fe05264d64324ca1f4436997efa1b/src/InputHandler.ts#L908-L918

Reply:
So glad I cc'd you on that tweet - this is so much better

Reply:
Doesn't seem to work for me on Windows.  Is support for the `3` option new? `\x1b[2J` works for me (in that it clears the current display) but `\x1b[3J` appears to be ignored.

Reply:
On Windows the compatibility layer might filter it out of something. I would expect this fix to be in the form of changing how clear acts in conpty to send ED 3 and 2 (i think) to clear the viewport and scrollback when `Console.Clear` is triggered. Maybe this is that issue? https://github.com/microsoft/terminal/issues/2260

Reply:
@SeeminglyScience were you trying:

```
Write-Host "`e[3J"
```

Reply:
Tried all of these:

```powershell
Write-Host "`e[3J"
$Host.UI.Write("`e[3J")
[Console]::Write("`e[3J")

Write-Host "$([char]0x1b)[3J"
$Host.UI.Write("$([char]0x1b)[3J")
[Console]::Write("$([char]0x1b)[3J")
```

Also switching `3` with `2` successfully clears the current screen, so it's definitely not a syntax problem.

Reply:
damn ok - thought that maybe this would be viable but we'll probably have to do what this PR does and use the vscode API.

Reply:
@SeeminglyScience one more thing:

```
$([char]27)
```

Reply:
@TylerLeonhardt What about it?

Reply:
I got this from one of the Clear issues in the new Windows Terminal.
```PowerShell
$host.UI.Write( "$([char]27)[3J$([char]27)[2J" )  
```

It mostly works but requires a few tweaks. As is, the command leaves the cursor at the bottom so I expanded on it to reset the cursor to the top:

```PowerShell
$host.UI.RawUI.CursorPosition = New-Object System.Management.Automation.Host.Coordinates 0,0
$host.UI.Write( "$([char]27)[3J$([char]27)[2J" )  
```

It seems to work better in latest preview. Last time I tested this, I was still having to do manual `Clear-Host` in addition to it.

Remaining issue is the scrollbar doesn't reset so it looks like there is stuff there but there isn't. I still use VS Code's Clear to remove it.

Reply:
Yeah VS Code's Clear is probably the most reliable here but that's great info @dragonwolf83!